src/AutoRest.CSharp/Common/Output/Expressions/KnownValueExpressions/AzureLocationExpression.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using Azure.Core;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace AutoRest.CSharp.Common.Output.Expressions.KnownValueExpressions
+{
+    internal sealed record AzureLocationExpression(ValueExpression Untyped) : TypedValueExpression<AzureLocation>(Untyped)
+    {
+    }
```

Comment:
This is an azure specific expression, we should put it in the `Azure` directory.
But I do not see anywhere we need this, we could omit this class until we actually need it.
Here "we need this" means we need to write expressions that access specific properties or methods from this type, but we are not actually needing those.